### PR TITLE
Refactor ngraph builders 3

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Remove or deprecate makeSqueezeUnsqueeze, makeMinMax, makeProposal, makeCumSum makeSelect, makeSpaceToDepth, makeShuffleChannels, makeMatMul builders
 - Suppress deprecation in legacy code. Remove usage in modern code

### Tickets:
 - [CVS-117139](https://jira.devtools.intel.com/browse/CVS-117139)

### CODE REVIEW

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/matmul_sparse.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ class MatMulSparseCPUTest : public testing::WithParamInterface<MatMulSparseParam
         auto matrixBFP32 = builder::makeDynamicInputLayer(element::f32, helpers::InputLayerType::CONSTANT, inShapeB);
 
         auto matMulRelaxed = std::make_shared<ov::op::TypeRelaxed<opset3::MatMul>>(
-            *as_type_ptr<opset3::MatMul>(builder::makeMatMul(inputParamsFP32, matrixBFP32, transpose_a, transpose_b)),
+            *std::make_shared<ov::op::v0::MatMul>(inputParamsFP32, matrixBFP32, transpose_a, transpose_b),
```
Comment (t-jankowski) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/matmul_sparse.cpp:136]: `shared_ptr` is not needed here.

File: src/core/tests/type_relaxed_copy.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@ TEST_F(TypeRelaxedThreading, TypeRelaxedCloning) {
     auto inp2 = std::make_shared<op::v0::Parameter>(element::i8, PartialShape{-1, -1, -1, -1});
 
     auto matMulRelaxed = std::make_shared<ov::op::TypeRelaxed<ov::op::v0::MatMul>>(
-        *as_type_ptr<op::v0::MatMul>(ngraph::builder::makeMatMul(inp1_f32, inp2_f32, false, false)),
+        *std::make_shared<ov::op::v0::MatMul>(inp1_f32, inp2_f32, false, false),
```
Comment (t-jankowski) [inline src/core/tests/type_relaxed_copy.cpp:45]: `shared_ptr` is not needed here.
```suggestion
        ov::op::v0::MatMul{inp1_f32, inp2_f32, false, false},
```

File: src/tests/functional/shared_test_classes/src/single_layer/space_to_depth.cpp [hunk 3]
```diff
@@ -46,7 +46,9 @@ void SpaceToDepthLayerTest::SetUp() {
     std::tie(inShape, inputPrecision, mode, blockSize, targetDevice) = this->GetParam();
     auto inPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(inputPrecision);
     ov::ParameterVector params {std::make_shared<ov::op::v0::Parameter>(inPrc, ov::Shape(inShape))};
-    auto s2d = ngraph::builder::makeSpaceToDepth(params[0], mode, blockSize);
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    auto s2d = std::make_shared<ov::op::v0::SpaceToDepth>(params[0], mode, blockSize);
+    OPENVINO_SUPPRESS_DEPRECATED_END
```
Comment (t-jankowski) [inline src/tests/functional/shared_test_classes/src/single_layer/space_to_depth.cpp:51]: Why is it marked as deprecated?

File: src/tests/functional/shared_test_classes/src/single_layer/space_to_depth.cpp [hunk 4]
```diff
@@ -46,7 +46,9 @@ void SpaceToDepthLayerTest::SetUp() {
     std::tie(inShape, inputPrecision, mode, blockSize, targetDevice) = this->GetParam();
     auto inPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(inputPrecision);
     ov::ParameterVector params {std::make_shared<ov::op::v0::Parameter>(inPrc, ov::Shape(inShape))};
-    auto s2d = ngraph::builder::makeSpaceToDepth(params[0], mode, blockSize);
+    OPENVINO_SUPPRESS_DEPRECATED_START
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/src/single_layer/space_to_depth.cpp:49]: no builder is used here

**Review by vurusovs — APPROVED** [review]
LGTM, comment may be not fixed because the file is going to be deleted

File: src/tests/functional/shared_test_classes/src/single_layer/space_to_depth.cpp (hunk 3 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/single_layer/space_to_depth.cpp:51]: I just made one change and then another and forgot to remove this macroses.

I will fix it in the next PR

**Pull request merged**